analysis/src/CreateInterface.ml

```diff
@@ -31,6 +31,82 @@ module SourceFileExtractor = struct
       !res)
 end
 
+module AttributesUtils : sig
+  type t
+
+  val make : string list -> t
+
+  val contain : string -> t -> bool
+
+  val toString : t -> string
+end = struct
+  type pos = int * int
+  type t = (pos * string) list
+  type parseState = Search | Collect of int
+
+  let getLineIdx ((lineIdx, _), _) = lineIdx
+
+  let getAttrOffset ((_, attrOffset), _) = attrOffset
+
+  let getAttrName (_, attribute) = attribute
+
+  let makeAttr lineIdx attrOffsetStart attrOffsetEnd line =
+    ( (lineIdx, attrOffsetStart),
+      String.sub line attrOffsetStart (attrOffsetEnd - attrOffsetStart) )
```

Comment:
Nit: to know that this is safe (not out of bound) you need to look at the callers of this function.
But if this function were to be moved into `make`, which is the only user, then the evidence that this is safe would be right there.